src/app/core/services/auth.service.ts

```diff
@@ -9,35 +13,50 @@ export class AuthService {
     isAuthenticated$ = new BehaviorSubject<boolean>(false);
     user$ = new BehaviorSubject<User>(null);
 
-    constructor() {
-        const userInfo = this.getUserInfo();
-        const isAuthenticated = userInfo ? userInfo.isAuthenticated : false;
-        this.isAuthenticated$.next(isAuthenticated);
-        this.user$.next(userInfo);
+    constructor(private httpService: HttpClient) {
+        this.isAuthenticated$.next(this.isUserAuthenticated());
+        this.user$.next(this.getUserInfo());
     }
 
     login(user: User) {
-        localStorage.setItem("userInfo", JSON.stringify(user));
-        this.isAuthenticated$.next(true);
-        this.user$.next(this.getUserInfo());
+        return this.httpService.post(APP_URLS.API_LOGIN, user, { observe: "response" }).pipe(
+            tap(({ status, body }: any) => {
+                if (status === 200) {
+                    localStorage.setItem("token", body.token);
+                    localStorage.setItem("isAuthenticated", "true");
+                    this.isAuthenticated$.next(this.isUserAuthenticated());
+                }
+            }),
+            map(({ status }) => ({
+                status
+            }))
+        );
     }
 
     logout() {
-        localStorage.removeItem("userInfo");
+        localStorage.clear();
         this.isAuthenticated$.next(false);
     }
 
-    getUserInfo(): User {
+    getUserInfo() {
         const userInfo = localStorage.getItem("userInfo");
         return userInfo ? JSON.parse(userInfo) : null;
     }
 
-    isAuthenticated() {
-        return this.isAuthenticated$.value;
+    isUserAuthenticated() {
+        const _isAuthenticated = localStorage.getItem("isAuthenticated");
+        return _isAuthenticated ? JSON.parse(_isAuthenticated) : false;
     }
 
-    isUserAuthenticated() {
-        const userInfo = this.getUserInfo();
-        return userInfo ? userInfo.isAuthenticated : false;
+    fetchUserDetails() {
```

Comment:
should return an observable. 
Creating subscription in service's method you create a memory leak every time you call method.
You should return ONLY observable from service's methods. Or the method should be void